shopinvader/services/cart.py

```diff
@@ -448,6 +448,7 @@ def _get(self, create_if_not_found=True):
 
     def _create_empty_cart(self):
         vals = self._prepare_cart()
+        # TODO: we should play onchanges here!!!
```

Comment:
any idea why onchanges are not called here? This can lead to a lot of mismatches between orders created via std odoo and orders created via shop.
I didn't want to change current behavior w/out discussing, but it seems broken to me.

CC @sebastienbeau

Reply:
> any idea why onchanges are not called here? This can lead to a lot of mismatches between orders created via std odoo and orders created via shop.
> I didn't want to change current behavior w/out discussing, but it seems broken to me.
> 
> CC @sebastienbeau

That's done in _prepare_cart : https://github.com/shopinvader/odoo-shopinvader/blob/13.0/shopinvader/services/cart.py#L466

:smile:

Reply:
@rousseldenis then it seems it does not work properly in some cases.
I had to call `onchange_partner_id` explicitly after creation in a custom module to fix it :man_shrugging: 
I'll add a test.

Reply:
Test added, it works.... I dunno WTH went on our instance then :thinking: